frontend/src/components/Navbar/NavBar.tsx

```diff
@@ -47,55 +60,56 @@ const Navbar: React.FC<NavbarProps> = ({ setIsCollapsed }) => {
                 <ul className="navbar-nav flex-row align-items-center gap-4">
                     {/* Feedback Panel Button */}
                     <li className="nav-item">
-                        <button className="btn btn-light btn-sm d-flex align-items-center gap-1">
+                        <button onClick={handleShowFeedbackRatingPanel} className="btn btn-light btn-sm d-flex align-items-center gap-1">
                             <IconMessageCircle className={styles.iconLarge} />
-                            <span className="d-none d-md-inline">Show feedback panel</span>
+                            <span className="d-none d-md-inline">{feedbackContent}</span>
                         </button>
                     </li>
 
                     {/* Hide Chat History Button */}
                     <li className="nav-item">
-                        <button className="btn btn-light btn-sm d-flex align-items-center gap-1">
+                        <button onClick={handleShowHistoryPanel} className="btn btn-light btn-sm d-flex align-items-center gap-1">
                             <IconHistory className={styles.iconLarge} />
-                            <span className="d-none d-md-inline">Hide chat history</span>
+                            <span className="d-none d-md-inline">{historyContent}</span>
                         </button>
                     </li>
 
                     {/* Settings Button */}
                     <li className="nav-item">
-                        <button className="btn btn-light btn-sm d-flex align-items-center gap-1">
+                        <button onClick={handleShowSettings} className="btn btn-light btn-sm d-flex align-items-center gap-1">
                             <IconSettings className={styles.iconLarge} />
                             <span className="d-none d-md-inline">Settings</span>
                         </button>
                     </li>
 
                     {/* User Profile Card */}
                     <li className="nav-item dropdown">
-                        <a className="nav-link" href="#" id="drop2" data-bs-toggle="dropdown" aria-expanded="false">
+                        <a className={`nav-link ${isDropdownOpen ? 'show' : ''}`} role="button" id="drop2" data-bs-toggle="dropdown" aria-expanded={isDropdownOpen} onClick={handleOnClickProfileCard}>
```

Comment:
@egdagger, if this element is intended to behave as a button, we might want to replace it with a <button> element to improve accessibility and semantic HTML, and adjust the style to make it look the same.